### DISCUSSION TITLE
Implementation Questions

### DISCUSSION THREAD

**Discussion by jpsla94** [magma/magma#6312] [Q&A] [answered]
Hello All,

I am currently developing a master thesis focused on PCRF and PCEF for applying control policies to devices based on their IMSI. And I found that this solution fits the bill for applying a open-source PCRF and PCEF in my master thesis.

And for that I have some questions to ask regrading this solution:

1. What packet inspection capabilities does it possess? Can it detect protocols(like http, rtp) and content types (like audio, video or ext) for example? And write policies based on this?
2. How are policies associated with the subscriber? How do I specify the attribution of a policy to a subscriber?
3. How can the alarms be delivery to the user? Email, Text Message? Can it generate and send reports of the policies that were violated by the device?

Thanks in advance.

**Comment by karthiksubraveti** [comment, marked as answer]
Looks like this thread addresses these issues https://magmacore.slack.com/archives/C0191LLL03S/p1619433029053800
